dashboard/app/controllers/api/v1/peer_review_submissions_controller.rb

```diff
@@ -61,13 +63,26 @@ def report_csv
     script = Plc::CourseUnit.find(params[:plc_course_unit_id]).script
     peer_reviewable_levels = ScriptLevel.where(script: script).select {|sl| sl.level.try(:peer_reviewable?)}.map(&:level).uniq
 
+    # We create peer reviews whenever an eligible submission occurs, so their creation timestamps
+    # are our best source of truth for submission dates.
+    # Here, gather all the submission dates we need in one query and organize them for easy
+    # access while generating the report.
+    submission_times_by_user_script_level = PeerReview.
+      where(submitter_id: enrollments.map(&:user_id)).
+      pluck(:submitter_id, :script_id, :level_id, :created_at).
+      group_by {|pr| pr[0..2]}. # group by [submitter_id, script_id, level_id]
+      map {|k, prs| [k, prs.map {|pr| pr[3]}]}. # only keep :created_at in values
+      to_h
+
     enrollments.each do |enrollment|
       peer_review_submissions = Hash.new
 
       UserLevel.where(user: enrollment.user, level: peer_reviewable_levels).each do |user_level|
+        submission_times = submission_times_by_user_script_level[[user_level.user_id, user_level.script_id, user_level.level_id]]
         peer_review_submissions[user_level.level.name] = {
           status: result_to_status(user_level.best_result),
-          date: user_level.created_at.strftime("%-m/%-d/%Y")
+          first_submission_date: submission_times.min&.strftime("%-m/%-d/%Y"),
+          last_submission_date: submission_times.max&.strftime("%-m/%-d/%Y")
```

Comment:
Notably, this comment in the spec...

> In the current CSV, we show the date for the most recent submission.

...was wrong.  Technically, we were showing the `UserLevel` creation date which is usually the date of the first submission (not the most recent), and in theory could be created before a submission actually occurs.

I've switched to using `PeerReview` creation dates as our authority for this information.  `PeerReview`s are created on submit or resubmit in [this `UserLevel` `after_submit` hook](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user_level.rb#L110-L125) making them a more reliable record of _submission_ than anything on `UserLevel`. That does add one somewhat complex query to this route.